File: Updog.Client/src/post/use-cases/find-by-user/post-finder-by-user.ts
import { ApiInteractor } from '@/core/api-interactor';
import { PaginationParams } from '@/core/pagination/pagination-params';
import { Post } from '@/post/domain/post';
import { PostMapper } from '@/post/infrastructure/post-mapper';
import { PostApiInteractor } from '@/post/infrastructure/post-api-interactor';
import { PostFinderByUserParams } from './post-finder-by-user-params';
import { PagedResultSet } from '@/core/pagination/paged-result-set';

/**
 * API interactor to find posts by the user that created them.
 */
export class PostFinderByUser extends PostApiInteractor<PostFinderByUserParams, PagedResultSet<Post>> {
    public async handle(input: PostFinderByUserParams): Promise<PagedResultSet<Post>> {
        const response = await this.http.get<Post[]>(`/post/user/${input.username}`, { params: input.paginationInfo });

        const pagination = this.getPaginationInfo(response);
        const posts = response.data.map((postInfo: any) => this.postMapper.map(postInfo));

        return new PagedResultSet(posts, pagination);
    }
}

File: Updog.Application/Post/UseCases/FindByUser/PostFinderByUser.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Updog.Application.Paging;
using Updog.Domain;

namespace Updog.Application {
    /// <summary>
    /// Use case handler to find a number of posts via the user who created them..
    /// </summary>
    public sealed class PostFinderByUser : Interactor<FindByValueParams<string>, PagedResultSet<PostView>> {
        #region Fields
        private IDatabase database;
        private IPostViewMapper postMapper;
        #endregion

        #region Constructor(s)
        public PostFinderByUser(IDatabase database, IPostViewMapper postMapper) {
            this.database = database;
            this.postMapper = postMapper;
        }
        #endregion


        #region Publics
        [Validate(typeof(FindByUserValidator))]
        protected override async Task<PagedResultSet<PostView>> HandleInput(FindByValueParams<string> input) {
            using (var connection = database.GetConnection()) {
                IPostRepo postRepo = database.GetRepo<IPostRepo>(connection);
                IVoteRepo voteRepo = database.GetRepo<IVoteRepo>(connection);

                PagedResultSet<Post> posts = await postRepo.FindByUser(input.Value, input.Pagination?.PageNumber ?? 0, input.Pagination?.PageSize ?? Post.PageSize);

                if (input.User != null) {
                    foreach (Post p in posts) {
                        p.Vote = await voteRepo.FindByUserAndPost(input.User.Username, p.Id);
                    }
                }

                return new PagedResultSet<PostView>(posts.Items.Select(p => postMapper.Map(p)), posts.Pagination);
            }
        }
    }
    #endregion
}